frontend/dockerfile/dockerfile2llb/convert_test.go

```diff
@@ -27,6 +27,57 @@ func toEnvMap(args []instructions.KeyValuePairOptional, env []string) map[string
 	return m
 }
 
+func TestDockerfileLinting(t *testing.T) {
+	t.Parallel()
+	df := `FROM scratch AS foo
+ENV FOO bar
+FROM foo
+COPY --from=foo f1 /
+	`
+	convertOpt := ConvertOpt{}
+	warningList, err := DockerfileLint(appcontext.Context(), []byte(df), convertOpt)
+	assert.NoError(t, err)
+	assert.Empty(t, warningList)
+
+	df = `from scratch AS foo
+env FOO bar
+from foo
+copy --from=foo f1 /
+	`
+	warningList, err = DockerfileLint(appcontext.Context(), []byte(df), convertOpt)
+	assert.NoError(t, err)
+	assert.Empty(t, warningList)
+
+	df = `FROM scratch AS FOO
+ENV FOO bar
```

Comment:
Oh! And if we consider "best practices", we could consider that a Dockerfile should start with a `# syntax`☺️

Reply:
Those are great suggestions! There's quite a few linting rules that we want to add moving forward, these three are meant to be just the initial set, with other linting rules being introduced in future PRs! I've updated the initial comment to note that, because this is definitely just a first stepping stone! :D

Reply:
For sure! I saw your PR and was interested; consider my comment purely as "hey, these might be interesting to add as well!"

Reply:
These are great, thanks @thaJeztah. I will send out a proposal shortly to capture more input on this.